### PR TITLE
🌱 Cleanup v1 cluster before creating v3 clusters on upgrade

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
**What this PR does / why we need it**:

Perform automatic provisioning cluster removal on upgrade with `v3 clusters` feature flag enabled.
In case of a duplicate cluster entries, a user can re-run the upgrade without settings change.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

### CODE REVIEW

**Review by alexander-demicev — APPROVED** [review]

**Review by salasberryfin — APPROVED** [review]
Thanks @Danil-Grigorev. LGTM

**Review by furkatgofurov7 — COMMENTED** [review]
LGTM, few nits below:

File: charts/rancher-turtles/templates/post-upgrade-job.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    namespace: rancher-turtles-system
+roleRef:
+  kind: ClusterRole
+  name: post-upgrade-job-delete-clusters
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: post-upgrade-delete-clusters
```
[the remark this answers is not in the document]
Comment (Danil-Grigorev) [reply charts/rancher-turtles/templates/post-upgrade-job.yaml:47]: I’d like to keep the name as is. This hook may be needed to provide downgrade functionality during feature stabilization. ADR will explain more.

File: test/e2e/specs/migrate_gitops_provv2_mgmtv3.go [hunk 2]
```diff
@@ -0,0 +1,415 @@
+//go:build e2e
```
[the remark this answers is not in the document]
Comment (Danil-Grigorev) [reply test/e2e/specs/migrate_gitops_provv2_mgmtv3.go:1]: e2e tests have this tag across all packages, this including.

File: charts/rancher-turtles/questions.yml [hunk 3]
```diff
@@ -53,6 +53,12 @@ questions:
     type: boolean
     label: Use management v3 cluster manifest
     group: "Rancher Turtles Features Settings"
+  - variable: rancherTurtles.features.managementv3-cluster-migration.enabled
+    default: false
+    description: "(Experimental) Automatically migrate provisioning clusters to v3 management clusters on upgrade"
+    type: boolean
+    label: Use newly provisioned management v3 cluster manifest, replacing provisioning v2
```
[the remark this answers is not in the document]
Comment (Danil-Grigorev) [reply charts/rancher-turtles/questions.yml:60]: I think it is best to remove the numbers. We are talking about API groups, not versions. I’ll update it shortly.

**Review by alexander-demicev — APPROVED** [review]

File: test/e2e/specs/migrate_gitops_provv2_mgmtv3.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (furkatgofurov7) [reply test/e2e/specs/migrate_gitops_provv2_mgmtv3.go:1]: I was referring to file name here actually. Also, what is v2 in this context? Is it same as provisioning v1 or similar or same thing?

[the remark this answers is not in the document]
Comment (Danil-Grigorev) [reply test/e2e/specs/migrate_gitops_provv2_mgmtv3.go:1]: Can we address this in a followup cleanup PR instead? @furkatgofurov7

**Review by furkatgofurov7 — APPROVED** [review]

**Pull request merged**